### PR TITLE
feat: Home page [#11]

### PR DESCRIPTION
**Pull request by threethingsapp**
**Background**
this is to add Home page.

**Change**
add Home screen and necessary components.

**Test Plan**

screen:

Screenshot 2023-12-29 at 00 32 06

design

Screenshot 2023-12-13 at 01 23 55

### CODE REVIEW

File: app/components/Button/Button.module.css [hunk 1]
```diff
@@ -3,22 +3,37 @@
...
 }
 
 .outline-button {
   border: 1px solid #4f4f4f;
+}
+
+.clear-button {
+  border: unset;
+  width: unset;
+  color: rgb(79, 79, 79);
```
Comment (threethingsapp) [inline app/components/Button/Button.module.css]: could we add some const for this colors? in hexa would be nice!

File: app/components/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+"use client";
```
Comment (threethingsapp) [inline app/components/Button/Button.tsx]: do we need this?

File: app/components/CategoryCard/CategoryCard.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  const isCompleted = tasks.completed === tasks.total;
+
+  return (
+    <Container className={styles.wrapper}>
+      <div className={styles["header-container"]}>
+        <Pill color={CategoryColors[category]}>
+          <Icon name={category} width={14} height={14} />
+        </Pill>
+
+        <div className={styles["avatars-container"]}>
```
Comment (threethingsapp) [inline app/components/CategoryCard/CategoryCard.tsx]: it would be cool if this were a component

File: app/components/Icon.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  email = "/email.svg",
+  plus = "/plus.svg",
+  home = "/home.svg",
+  notepad = "/notepad.svg",
+}
+
+const DEFAULT_SIZE = 24;
+
+type Props = {
+  name: 'search' | 'books' | 'work' | 'urgent' | 'email' | 'plus' | 'home' | 'notepad';
```
Comment (threethingsapp) [inline app/components/Icon.tsx]: it could be a `enum`

File: app/components/RandomPeople/RandomPeople.module.css [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+.img15 {
+  background-position-x: calc((calc((var(--image-size)) * 2) + (var(--leftGap) * 3)) * -1) ;
+  background-position-y: calc((calc(var(--image-size) * 3) + calc(var(--topGap) * 6)) * -1);
+}
+
+.img16 {
+  background-position-x: calc((calc((var(--image-size)) * 3) + (var(--leftGap) * 4)) * -1) ;
+  background-position-y: calc((calc(var(--image-size) * 3) + calc(var(--topGap) * 6)) * -1);
+}
```
Comment (threethingsapp) [inline app/components/RandomPeople/RandomPeople.module.css]: could we do some iteration here to reduce the code?

File: app/home/page.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+      <div style={{ height: "140px" }}>
+        <DailyCard className={css(styles.card, styles["first-card"])} />
+        <DailyCard className={css(styles.card, styles["second-card"])} />
+        <DailyCard className={css(styles.card, styles["third-card"])} />
+      </div>
+
+      <Heading variant="v500">Categories</Heading>
+
+      <div className={styles["category-container"]}>
```
Comment (threethingsapp) [inline app/home/page.tsx:37]: I think we can have a list with the categories data and intereate it to create this components.

File: app/home/home.module.css [hunk 7]
```diff
@@ -0,0 +1,45 @@
+.button-wrapper {
+  display: flex;
+}
+
+.card {
+  position: relative;
+  width: 424px;
+  box-shadow: rgba(9, 30, 66, 0.25) 0px 1px 1px, rgba(9, 30, 66, 0.13) 0px 0px 1px 1px;
```
Comment (threethingsapp) [inline app/home/home.module.css:10]: maybe we can use some color variables here.

**Comment by threethingsapp** [comment]
general comments about design

- the Overview / Productivity shouldn't be centered. 
- daily progress card looks more wider.
- Category title should be a bit more bigger and add some space on top.
- category card should be more squeared, grow up the pill icon, remove some inside spaces and remove the percentage in the progress bar

File: app/components/Button/Button.module.css (hunk 1 above)
Comment (threethingsapp) [reply app/components/Button/Button.module.css]: done!

File: app/components/Button/Button.tsx (hunk 2 above)
Comment (threethingsapp) [reply app/components/Button/Button.tsx]: done!

File: app/components/CategoryCard/CategoryCard.tsx (hunk 3 above)
Comment (salvagit) [reply app/components/CategoryCard/CategoryCard.tsx]: done!

File: app/components/Icon.tsx (hunk 4 above)
Comment (salvagit) [reply app/components/Icon.tsx]: done!

File: app/components/RandomPeople/RandomPeople.module.css (hunk 5 above)
Comment (salvagit) [reply app/components/RandomPeople/RandomPeople.module.css]: done!

File: app/home/page.tsx (hunk 6 above)
Comment (salvagit) [reply app/home/page.tsx:37]: done!

**Pull request merged**